lib/migrations/createGenesisDao.ts

```diff
@@ -23,135 +29,130 @@ interface FounderSpec {
    */
   reputation: string | number;
 }
+
+export interface ForgedDaoInfo {
+  avatarAddress: Address;
+  orgName: string;
+  daoCreatorInst: any;
+}
+
 /**
  * Migration callback
  */
-export const arcJsDeployer = (web3: Web3, artifacts: any, deployer: any): void => {
-
-  // so Utils.getWeb3 can find it
-  (global as any).web3 = web3;
-
-  const network = env.arcjs_network || "ganache";
-
-  const internalFoundersConfigLocation = "../../migrations/founders.json";
-  const foundersConfig = require(internalFoundersConfigLocation).founders;
+export class GenesisDaoCreator {
 
-  const customFoundersConfigLocation = env.arcjs_foundersConfigurationLocation || internalFoundersConfigLocation;
-
-  if (internalFoundersConfigLocation !== customFoundersConfigLocation) {
-    console.log(`merging custom founders from ${customFoundersConfigLocation}`);
-    const customFoundersConfig = require(customFoundersConfigLocation).founders;
-    // merge the two
-    Object.assign(foundersConfig, customFoundersConfig);
-  }
-
-  const founders = foundersConfig[network];
-
-  if (!founders || (founders.length === 0)) {
-    throw new Error(`no founders were given for the network: ${network}`);
+  constructor(
+    private web3: Web3,
+    private network: string) {
   }
 
   /**
-   * Truffle Solidity artifact wrappers
+   * Forge the Genesis DAO.  Note this does not set the schemes.
    */
-  const AbsoluteVote = artifacts.require("AbsoluteVote.sol");
-  const Avatar = artifacts.require("Avatar.sol");
-  const ContributionReward = artifacts.require("ContributionReward.sol");
-  const DaoCreator = artifacts.require("DaoCreator.sol");
-  // ExecutableTest is used only by tests
-  const ExecutableTest = artifacts.require("ExecutableTest.sol");
-  const GlobalConstraintRegistrar = artifacts.require("GlobalConstraintRegistrar.sol");
-  const SchemeRegistrar = artifacts.require("SchemeRegistrar.sol");
-  const SimpleICO = artifacts.require("SimpleICO.sol");
-  const TokenCapGC = artifacts.require("TokenCapGC.sol");
-  const UController = artifacts.require("UController.sol");
-  const UpgradeScheme = artifacts.require("UpgradeScheme.sol");
-  const VestingScheme = artifacts.require("VestingScheme.sol");
-  const VoteInOrganizationScheme = artifacts.require("VoteInOrganizationScheme.sol");
-  const GenesisProtocol = artifacts.require("GenesisProtocol.sol");
-  const ControllerCreator = artifacts.require("ControllerCreator.sol");
+  public async forge(foundersConfigurationLocation: string): Promise<ForgedDaoInfo> {
```

Comment:
How is calling this function ?
How about adding module.exports.createDao =function(params...) {
...do the dao creation...
}
then it could be easily be executed externaly from the command line with something like : 
`node -e 'require("./createDao.js").createDao(params..))'`

Reply:
`forge` and `setSchemes` are called from /package-scripts/createGenesisDao.js which is invoked from package-scripts.js.

Regarding module.exports, we already have the `DAO` class on which applications can call `DAO.new` to create a DAO given a configuration object.

I am not using `DAO.new` in the migration script mainly I want to minimize the complexity of the script, and there has been no request to support the ability to receive arbitrary DAO configurations, other than Genesis.

In any case, I would prefer to use a separate PR for creating a script to support adding arbitrary DAOs.